src/logics/Logic.cc

```diff
@@ -1564,6 +1570,30 @@ lbool Logic::retrieveSubstitutions(const vec<PtAsgn>& facts, Map<PTRef,PtAsgn,PT
     return l_Undef;
 }
 
+void Logic::substitutionsTransitiveClosure(Map<PTRef, PtAsgn, PTRefHash> & substs) {
+    bool changed = true;
+    vec<PTRef> keys;
+    substs.getKeys(keys);
```

Comment:
Since varsubstitute does not change the Map, this is indeed possible and should be more efficient.
I have changed the code according to the suggestion and added a unit test.